### PR TITLE
feat: Adds a config.gatherStats parameter.

### PR DESCRIPTION
**Pull request by gpolitis**
This PR exposes the stats gathering functionality that's already built-in the TraceablePeerConnection that has it gather stats from the wrapped PeerConnection. This is useful in debug scenarios as it allows to extract and process the stats.

Please don't merge this yet, it's WIP and I haven't tested it yet.

### CODE REVIEW

File: modules/xmpp/JingleSessionPC.js [hunk 1]
```diff
@@ -248,6 +248,10 @@ export default class JingleSessionPC extends JingleSession {
 
         const pcOptions = { disableRtx: this.room.options.disableRtx };
 
+        if (this.room.options.gatherStats) {
+            pcOptions.maxstats = 300;
```
Comment (bbaldino) [inline modules/xmpp/JingleSessionPC.js]: can you add a constant for this? `DEFAULT_MAX_STATS` or something?

File: modules/RTC/TraceablePeerConnection.js [hunk 2]
```diff
@@ -185,9 +185,10 @@ export default function TraceablePeerConnection(
     this.statsinterval = null;
 
     /**
-     * @type {number}
+     * @type {number} The max number of stats to keep in this.stats. Limit to
+     * 300 values, i.e. 5 minutes; set to 0 to disable
      */
-    this.maxstats = 0;
+    this.maxstats = options.maxstats;
```
Comment (virtuacoplenny) [inline modules/RTC/TraceablePeerConnection.js]: Is there a guarantee config.gatherStats will be defined? Should there be a default set somewhere?

Comment (gpolitis) [reply modules/RTC/TraceablePeerConnection.js]: config.gatherStats doesn't need to be defined, falsy is okay for this.maxstats.

**Comment by gpolitis** [comment]
This PR is now ready for review. It's a few-liner, @bbaldino @virtuacoplenny or anyone else, please feel free to merge it/approve it/comment on it.

**Comment by gpolitis** [comment]
test this please

**Comment by gpolitis** [comment]
@virtuacoplenny, @bbaldino could you please take a look?

**Pull request merged**